src/v/cloud_storage/partition_manifest.cc

```diff
@@ -2008,6 +2100,49 @@ void partition_manifest::serialize_replaced(
     }
     cursor->replaced_done = true;
 }
+void partition_manifest::serialize_spillover(
+  partition_manifest::serialization_cursor_ptr cursor) const {
+    auto& w = cursor->writer;
+    if (_spillover_manifests.empty()) {
+        cursor->next_spill_offset = model::offset{};
+        cursor->spillover_done = true;
+        return;
+    }
+
+    if (
+      cursor->next_spill_offset == model::offset{}
+      && !_spillover_manifests.empty()) {
+        // The method is called first time
+        w.Key("spillover");
+        w.StartObject();
+        cursor->next_spill_offset = _spillover_manifests.begin()->base_offset;
+    }
+
+    if (!_spillover_manifests.empty()) {
+        auto it = _spillover_manifests.lower_bound(cursor->next_spill_offset);
+        for (; it != _spillover_manifests.end(); ++it) {
+            serialize_segment_meta(*it, cursor);
```

Comment:
nit: it's a bit weird that we use the segment name format for spillover manifest keys, but it doesn't really matter.

Reply:
agreed, will change that